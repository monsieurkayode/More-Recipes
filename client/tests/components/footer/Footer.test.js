/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '../../../src/components/footer';

const wrapper = mount(<Footer />);
describe('The Footer Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render and have a footer element', () => {
    expect(wrapper.find('footer').length).toBe(1);
  });

  it('should have an unordered list element', () => {
    expect(wrapper.find('ul').length).toBe(1);
  });

  it('should have four list item child elements', () => {
    expect(wrapper.find('ul').children().length).toBe(4);
  });

  it('should have li elements containing a span element each', () => {
    expect(wrapper.find('li').first().find('span').length).toBe(1);
  });

  it('should have a copyright footer text at the bottom', () => {
    expect(wrapper.find('.container').text())
      .toBe('Â© Andela 2018Created by Kayode Adeola');
  });
});
