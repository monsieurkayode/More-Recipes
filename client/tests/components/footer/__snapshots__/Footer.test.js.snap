// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Footer Component should render correctly 1`] = `
<footer
  className="page-footer blue-grey darken-2"
>
  <div
    className="row"
  >
    <div
      className="col l3 social-icon-container"
    >
      <ul>
        <li>
          <div>
            <span>
              <i
                className="fa fa-2x fa-github teal-text"
              />
            </span>
          </div>
        </li>
        <li>
          <div>
            <span>
              <i
                className="fa fa-2x fa-facebook teal-text"
              />
            </span>
          </div>
        </li>
        <li>
          <div>
            <span>
              <i
                className="fa fa-2x fa-twitter teal-text"
              />
            </span>
          </div>
        </li>
        <li>
          <div>
            <span>
              <i
                className="fa fa-2x fa-google teal-text"
              />
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="footer-copyright"
  >
    <div
      className="container"
    >
      Â© Andela 2018
      <a
        className="grey-text text-lighten-4 right"
        href="/"
      >
        Created by Kayode Adeola
      </a>
    </div>
  </div>
</footer>
`;
